taskvine/src/manager/vine_manager.c

```diff
@@ -765,6 +765,36 @@ static void cleanup_worker(struct vine_manager *q, struct vine_worker_info *w)
 	cleanup_worker_files(q, w);
 }
 
+/* Recover and replicate temp files of to be removed worker to reach threshold n */
+
+static void recover_worker_temp_files(struct vine_manager *q, struct vine_worker_info *w)
+{
+	char *cached_name = NULL;
+	struct vine_file_replica *info = NULL;
+
+	// Iterate over files we want might want to recover
+	HASH_TABLE_ITERATE(w->current_files, cached_name, info)
+	{
+		struct vine_file *f = hash_table_lookup(q->file_table, cached_name);
+
+		if (f && f->type == VINE_TEMP) { // replicate temp files
+
+			struct set *workers = hash_table_lookup(q->file_worker_table, cached_name);
+
+			if (workers) {
+				set_remove(workers, w);
```

Comment:
This code seems to duplicate what's going on in `vine_file_replica_table_remove`, please make use of that function.